### ISSUE TITLE
[DataStore] Performing save() while user is unauthenticated will cause app to freeze

### ISSUE THREAD

**Issue by nubpro** [aws-amplify/amplify-js#6406]
**Describe the bug**
The mutation will infinitely retry if client performs save() when user is not authenticated.

**To Reproduce**
Two ways to reproduce this:
Method 1: (This method can reproduce the issue consistently)
1. Logout
2. Perform a `DataStore.save()`.
3. App will freeze because of infinite retry in mutation processor.

Method 2: (This is slightly difficult to reproduce but still doable)
1. Perform a `DataStore.save()`.
2. Quickly logout with or without calling `DataStore.clear()`.
3. App will freeze because of infinite retry in mutation processor.

**Expected behavior**
1. `DataStore.save()` should throw error if user is not authenticated.
2. Mutation processor should exit loop when GraphQL returns 'No current user', otherwise it stuck in `while(tryWith)` loop.

**Screenshots**
With debugging mode enabled, you'll notice the console is spammed with 'No current user' error as mutationProcessor infinitely try the mutation.
![image](https://user-images.githubusercontent.com/762914/88365405-7511d180-cdb8-11ea-9006-d2fc9c19effe.png)

**What is Configured?**
<details>
  <summary><strong>Environment</strong></summary>

```
  System:
    OS: macOS 10.15.5
    CPU: (4) x64 Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
    Memory: 189.99 MB / 8.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.3.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Browsers:
    Chrome: 84.0.4147.89
    Safari: 13.1.1
  npmPackages:
    uuid: 3.3.2 => 3.3.2 
  npmGlobalPackages:
    @aws-amplify/cli: 4.22.0
    ios-deploy: 1.10.0
    npm: 6.14.4
    react-native-cli: 2.0.1
```

</details>

**Smartphone (please complete the following information):**

- Device: iPad
- OS: iOS 13

**Comment by nubpro** [comment]
@manueliglesias This issue I raised is similar to #6278 as they are stuck in an infinite loop within the mutation processor.

**Comment by sammartinez** [comment]
@nubpro can you provide a little more detail on your use case? are you using public models with your CRUD operations? Can you also provide the code snippet you have with your methods you are testing against above? Thanks ahead of time

**Comment by nubpro** [comment]
> @nubpro can you provide a little more detail on your use case? are you using public models with your CRUD operations? Can you also provide the code snippet you have with your methods you are testing against above? Thanks ahead of time

My models are limited to authenticated user only.
`@auth(rules: [{ allow: owner }, { allow: groups, groups: ["Admin"] }]) {`

I believe u can reproduce this bug with the sample app (https://docs.amplify.aws/lib/datastore/examples/q/platform/js#example-application).

**Comment by iartemiev** [comment]
This should be fixed as part of the DataStore Multi-Auth work that was released in aws-amplify@3.4.0. Please let us know if you're still having this issue.

**Comment by nubpro** [comment]
> This should be fixed as part of the DataStore Multi-Auth work that was released in aws-amplify@3.4.0. Please let us know if you're still having this issue.

Ah, thats a relieve
I dont have any DataStore app anymore, so lets just close this since it is addressed like you said
